### PR TITLE
Add KAT and PCT tests for ED25519 and ED448.

### PR DESCRIPTION
**Pull request by jallisonciq**

### CODE REVIEW

File: providers/implementations/keymgmt/ecx_kmgmt.c [hunk 1]
```diff
@@ -612,11 +733,31 @@ static void *ecx_gen(struct ecx_gen_ctx *gctx)
...
         if (!ossl_ed448_public_from_private(gctx->libctx, key->pubkey, privkey,
                                             gctx->propq))
             goto err;
+#ifdef FIPS_MODULE
+        if (!fips_eddsa_pct(gctx->libctx,
+                            ECX_KEY_TYPE_ED448,
+                            key->pubkey,
+                            privkey,
+                            key->keylen)) {
+            /* rsa pct aborts on fail, do the same. */
```
Comment (jason-rodri) [inline providers/implementations/keymgmt/ecx_kmgmt.c:757]: comment states RSA should be EdDSA

File: providers/implementations/keymgmt/ecx_kmgmt.c [hunk 2]
```diff
@@ -612,11 +733,31 @@ static void *ecx_gen(struct ecx_gen_ctx *gctx)
         if (!ossl_ed25519_public_from_private(gctx->libctx, key->pubkey, privkey,
                                               gctx->propq))
             goto err;
+#ifdef FIPS_MODULE
+        if (!fips_eddsa_pct(gctx->libctx,
+                            ECX_KEY_TYPE_ED25519,
+                            key->pubkey,
+                            privkey,
+                            key->keylen)) {
+            /* rsa pct aborts on fail, do the same. */
```
Comment (jason-rodri) [inline providers/implementations/keymgmt/ecx_kmgmt.c:742]: https://github.com/ciq-rocky-fips/openssl/pull/12/files#r1829882352

File: providers/implementations/keymgmt/ecx_kmgmt.c [hunk 3]
```diff
@@ -612,11 +733,31 @@ static void *ecx_gen(struct ecx_gen_ctx *gctx)
...
+            abort();
```
Comment (jason-rodri) [inline providers/implementations/keymgmt/ecx_kmgmt.c:743]: should we abort? or should we return an error?

File: providers/fips/self_test_data.inc [hunk 4]
```diff
@@ -1733,6 +1812,20 @@ static const ST_KAT_SIGN st_kat_sign_tests[] = {
         ITM(rsa_expected_sig)
     },
 #ifndef OPENSSL_NO_EC
+    {
+        OSSL_SELF_TEST_DESC_SIGN_ED25519,
```
Comment (jason-rodri) [inline providers/fips/self_test_data.inc:1816]: should ED be under OPENSSL_NO_EC?

**Review by jason-rodri — CHANGES_REQUESTED** [review]
Please address the comments, otherwise it looks good

File: providers/implementations/keymgmt/ecx_kmgmt.c (hunk 1 above)
Comment (jallisonciq) [reply providers/implementations/keymgmt/ecx_kmgmt.c:757]: What I meant in the comment is: "In generate key, the RSA pct test calls abort() when it fails" - so I cause the eddsa PCT test to do the same. Sorry if this wasn't clear (I thought it was).

File: providers/implementations/keymgmt/ecx_kmgmt.c (hunk 3 above)
Comment (jallisonciq) [reply providers/implementations/keymgmt/ecx_kmgmt.c:743]: abort() is what the RSA PCT does in this case. I just made eddsa do the same.

File: providers/fips/self_test_data.inc (hunk 4 above)
Comment (jallisonciq) [reply providers/fips/self_test_data.inc:1816]: Well, the EDDSA code lives in the tree under: crypto/ec/curve/curve25519.c and crypto/ec/curve448/ so I think it *is* an EC curve. I didn't want to just arbitrarily include it, so this seemed the best #define to use.

There is no OPENSSL_NO_EDDSA define so I think openssl considers these curves as part of elliptic curve crypto.

**Review by jason-rodri — APPROVED** [review]

**Pull request merged**